src/server/req_stat.c

```diff
@@ -819,24 +819,15 @@ req_stat_sched(struct batch_request *preq)
 	CLEAR_HEAD(preply->brp_un.brp_status);
 
 	psched = NULL;
-	if(strlen(preq->rq_ind.rq_status.rq_id) != 0) {
-		psched = find_scheduler(preq->rq_ind.rq_status.rq_id);
-		if(psched) {
-			rc = status_sched(psched, preq, &preply->brp_un.brp_status);
-		} else {
-			req_reject(PBSE_UNKSCHED, 0, preq);
-			return;
-		}
-	} else {
-		psched = (pbs_sched *) GET_NEXT(svr_allscheds);
-		while (psched != (pbs_sched *) 0) {
-			rc = status_sched(psched, preq, &preply->brp_un.brp_status);
-			if (rc != 0) {
-				break;
-			}
-			psched = (pbs_sched *) GET_NEXT(psched->sc_link);
+	psched = (pbs_sched *) GET_NEXT(svr_allscheds);
```

Comment:
You're iterating, use a for() loop.

Reply:
Ok I will change that, but I dont think there will be any difference in the compiled binary code